include/deal.II/grid/tria_accessor.h

```diff
@@ -2489,6 +2496,126 @@ class CellAccessor :  public TriaAccessor<dim,dim,spacedim>
+   * is similar to the function neighbor_child_on_subface(). The
+   * implementation of this function is consistent with
+   * periodic_neighbor_of_coarser_periodic_neighbor(). For instance,
+   * assume that we are sitting on a cell named @c cell1, whose neighbor behind
+   * the @c ith face is one level coarser. Let us name this coarser neighbor
+   * @c cell2. Then, by calling
+   * periodic_neighbor_of_coarser_periodic_neighbor(), from @c cell1, we get
+   * a @c face_num and a @c subface_num. Now, if we call
+   * periodic_neighbor_child_on_subface() from cell2, with the above face_num
+   * and subface_num, we get an iterator to @c cell1.
+   */
+  TriaIterator<CellAccessor<dim, spacedim> >
+  periodic_neighbor_child_on_subface (const unsigned int face_no,
+                                      const unsigned int subface_no) const;
+
+  /**
+   * This function is a generalization of
+   * periodic_neighbor_of_periodic_neighbor()
+   * for those cells which have a coarser periodic neighbor. The returned
+   * pair of numbers can be used in periodic_neighbor_child_on_subface()
+   * to get back to the current cell. In other words, the following
+   * assertion should be true, for a cell with coarser periodic neighbor:
+   * cell->periodic_neighbor(i)->periodic_neighbor_child_on_subface(face_no, subface_no)==cell
+   */
+  std::pair<unsigned int, unsigned int>
+  periodic_neighbor_of_coarser_periodic_neighbor (const unsigned face_no) const;
+
+  /**
+   * This function returns the index of the periodic neighbor. If there is
+   * no periodic neighbor at the given face, the returned value is -1.
+   */
+  int
+  periodic_neighbor_index (const unsigned int i) const;
+
+  /**
+   * This function returns the level of the periodic neighbor. If there is
+   * no periodic neighbor at the given face, the returned value is -1.
+   */
+  int
+  periodic_neighbor_level (const unsigned int i) const;
+
+  /**
+   * For a cell with a periodic neighbor at its @c ith face, this function
+   * returns the face number of that periodic neighbor such that the
+   * current cell is the periodic neighbor of that neighbor. In other words
+   * the following assertion holds for those cells which have a periodic
+   * neighbor with the same or a higher level of refinement as the current
+   * cell:
+   * @c {cell->periodic_neighbor(i)->
+   *     periodic_neighbor(cell->periodic_neighbor_of_periodic_neighbor(i))==cell}
+   * For the cells with a coarser periodic neighbor, one should use
+   * periodic_neighbor_of_coarser_periodic_neighbor() and
+   * periodic_neighbor_child_on_subface()
+   * to get back to the current cell.
+   */
+  unsigned int
+  periodic_neighbor_of_periodic_neighbor (const unsigned int i) const;
+
+  /**
+   * If a cell has periodic neighbor, this function returns the face number
+   * of the neighbor, which is connected to this cell.
+   */
+  unsigned int
+  periodic_neighbor_face_no (const unsigned int i) const;
```

Comment:
Concerning `i` as the argument of `periodic_neighbor_face_no`, I have fixed the documentation in the following way:
```
If a cell has periodic neighbor at its @c ith face, this function
returns the face number of the periodic neighbor, which is
connected to the @c ith face of this cell.
 ```
Concerning the difference between `periodic_neighbor_face_no` and `periodic_neighbor_of_periodic_neighbor`, I checked the difference between `neighbor_face_no` and `neighbor_of_neighbor`. It looks to me that when the neighbor is coarser `neighbor_of_neighbor` throws an exception, while `neighbor_face_no` returns the first element of `neighbor_of_coarser_neighbor`. However, in the `Triangulation::periodic_face_map` Daniel stores all of the coarser periodic neighbors of each cell (Daniel can correct me if I am wrong). In this case, `periodic_neighbor_face_no` and `periodic_neighbor_of_periodic_neighbor` are the same. If I am wrong (i.e. we do not have all of the coarser neighbors of a cell in `Triangulation::periodic_face_map`), then I need to fix this issue.